src/deployment.rs

```diff
@@ -79,6 +85,118 @@ pub struct Artifact {
     pub kind: ArtifactKind
 }
 
+impl DeployWithServePath {
+    pub fn new(serve_path: &Option<String>) -> Result<Self, Error> {
+        use std::path::MAIN_SEPARATOR as SEP;
+
+        let serve_path = if let Some(ref path) = *serve_path {
+            let mut double_sep = SEP.to_string();
+            double_sep.push(SEP);
+
+            // As state in README.md
+            // serve-path is sub and relative to deploy-path
+            // It is not allow to contains `\\` or `..`
+            if path.contains( &double_sep ) || path.contains( ".." ) {
+                return Err( Error::ConfigurationError( format!("serve-path is invalid: {}", path) ) );
+            }
```

Comment:
The validity of these should all be checked when the config file is loaded.

Reply:
Yes, it is better to do this when loading config